src/utils/__tests__/documentdb.test.ts

```diff
@@ -671,3 +672,71 @@ describe("queryAttachments", () => {
     }
   });
 });
+
+describe("fromQueryEither", () => {
+  it("should return the correct query result", async () => {
+    interface IExpectedResultType {
+      foo: string;
+    }
+    const documentResult: IExpectedResultType = { foo: "bar" };
+    const queryFn: () => Promise<
+      Either<DocumentDb.QueryError, IExpectedResultType>
+    > = async () => right(documentResult);
+
+    const resultTE: TaskEither<
+      DocumentDbUtils.QueryError,
+      IExpectedResultType
+    > = DocumentDbUtils.fromQueryEither(queryFn);
+
+    const result = await resultTE.run();
+
+    expect(result.isRight()).toBeTruthy();
+    expect(result.value).toEqual(documentResult);
+  });
+
+  it("should return the correct query error on query left", async () => {
+    interface IExpectedResultType {
+      foo: string;
+    }
+    const cosmosdbError: DocumentDb.QueryError = {
+      body: "Bad Request",
+      code: 400
+    };
+    const queryFn: () => Promise<
+      Either<DocumentDb.QueryError, IExpectedResultType>
+    > = async () => left(cosmosdbError);
+
+    const resultTE: TaskEither<
+      DocumentDbUtils.QueryError,
+      IExpectedResultType
+    > = DocumentDbUtils.fromQueryEither(queryFn);
+
+    const result = await resultTE.run();
+
+    expect(result.isLeft()).toBeTruthy();
+    expect(result.value).toEqual(cosmosdbError);
+  });
+
+  it("should return the correct query error on rejection", async () => {
+    interface IExpectedResultType {
+      foo: string;
+    }
+
+    const errorMessage = "generic error";
+    const queryFn: () => Promise<
+      Either<DocumentDb.QueryError | Error, IExpectedResultType>
```

Comment:
It looks like this promise here (the queryfn) cannot never return `left(Error)`

```suggestion
      Either<DocumentDb.QueryError, IExpectedResultType>
```